### PR TITLE
Update MAFFT module with new arguments and tests

### LINKED ISSUE #4353: [FEATURE] MAFFT extend add sequences options
### Is your feature request related to a problem? Please describe

The module mafft has multiple ways of adding sequences, currently only one is supported the general '--add'.  https://mafft.cbrc.jp/alignment/server/add.html
The module should be slightly modified so other options can also be used. 

### Describe the solution you'd like

def args2 = task.ext.args2 ?: '--add'
def add  = addsequences ? "${args2} ${addsequences}" : ' '

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Joon-Klaps**
## PR checklist

Closes #4353 

- [X] This comment contains a description of changes (with reason).
- [X] If you've fixed a bug or added code that should be tested, add tests!
- [X] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [X] Follow the naming conventions.
- [X] Follow the parameters requirements.
- [X] Follow the input/output options guidelines.
- [X] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/mafft/tests/main.nf.test [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+                    [ id:'test', single_end:false ], // meta map
+                        file(params.test_data['sarscov2']['illumina']['scaffolds_fasta'], checkIfExists: true)
+                    ]
+                input[1] = []
+                """
+            }
+        }
+
+        then {
+            assert process.success
```
Comment (sateeshperi) [inline modules/nf-core/mafft/tests/main.nf.test:27]: assertAll()

File: modules/nf-core/mafft/main.nf [hunk 2]
```diff
@@ -20,8 +20,10 @@ process MAFFT {
 
     script:
     def args = task.ext.args ?: ''
+    def args2 = task.ext.args2 ?: '--add'
     def prefix = task.ext.prefix ?: "${meta.id}"
-    def add = addsequences ? "--add $addsequences" : ''
+    def add = addsequences ? "${args2} ${addsequences}" : ''
```
Comment (nvnieuwk) [inline modules/nf-core/mafft/main.nf:25]: The way it was done before was correct. If you don't want to use the `addsequences` files, you simply don't enter it into the module.

Comment (nvnieuwk) [reply modules/nf-core/mafft/main.nf:25]: I addition to that: `args2` should only be used for the arguments of a secondary tool used by the module and not for logic in the module :)

File: modules/nf-core/mafft/tests/main.nf.test [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        }
+
+    }
+
+    test("SARS-CoV-2 scaffolds fasta - add normal") {
+
+        when {
+            params {
+                outdir   = "${outputDir}"
+            }
```
[the remark this answers is not in the document]
Comment (Joon-Klaps) [reply modules/nf-core/mafft/tests/main.nf.test:38]: Aah, wasn't aware! Thanks for mentioning it!

File: modules/nf-core/mafft/main.nf (hunk 2 above)
Comment (Joon-Klaps) [reply modules/nf-core/mafft/main.nf:25]: https://github.com/nf-core/modules/issues/4353, there are multiple ways to add sequences https://mafft.cbrc.jp/alignment/server/add.html. 
It's not possible to use those other methods the way it was written previously.

**Review by vagkaratzas — COMMENTED** [review]
Great stuff. Maybe also update the container version since you are on it? A new one came out about a month ago: 7.520--h031d066_3

File: modules/nf-core/mafft/main.nf (hunk 2 above)
Comment (nvnieuwk) [reply modules/nf-core/mafft/main.nf:25]: IMO the best option would be to either that input a tuple with the file and a value that states which input it should be or to create an input channel for each option. The last option is the most nf-core-y way of doing it though :grin: (but I like the first one more)

**Comment by Joon-Klaps** [comment]
> Great stuff. Maybe also update the container version since you are on it? A new one came out about a month ago: 7.520--h031d066_3

The version already is 7.520, is there a specific reason for the new build?

**Comment by vagkaratzas** [comment]
> > Great stuff. Maybe also update the container version since you are on it? A new one came out about a month ago: 7.520--h031d066_3
> 
> The version already is 7.520, is there a specific reason for the new build?

I can't find any specific build details online I am afraid. Wouldn't be anything major probably.

File: modules/nf-core/mafft/main.nf (hunk 2 above)
Comment (nvnieuwk) [reply modules/nf-core/mafft/main.nf:25]: Oh yes if it's possible to do combinations of them all, you should definitely make them separate input channels (sorry, I know it seems weird :grin: ).

File: modules/nf-core/mafft/tests/main.nf.test [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+    script "../main.nf"
+    process "MAFFT"
+    tag "modules"
+    tag "modules_nfcore"
+    tag "mafft"
+
+    // test("SARS-CoV-2 scaffolds fasta") {
+    //     when {
+    //         process {
+    //             """
```
Comment (sateeshperi) [inline modules/nf-core/mafft/tests/main.nf.test:13]: why are these tests commented out ? @Joon-Klaps

File: modules/nf-core/mafft/main.nf [hunk 5]
```diff
@@ -4,12 +4,16 @@ process MAFFT {
...
+        'biocontainers/mafft:7.520--h031d066_3' }"
 
     input:
     tuple val(meta), path(fasta)
-    path  addsequences
+    tuple val(meta2),path(add)
+    tuple val(meta3),path(addfragments)
+    tuple val(meta4),path(addfull)
+    tuple val(meta5),path(addprofile)
+    tuple val(meta6),path(addlong)
```
Comment (sateeshperi) [inline modules/nf-core/mafft/main.nf:16]:
```suggestion
    tuple val(meta2), path(add)
    tuple val(meta3), path(addfragments)
    tuple val(meta4), path(addfull)
    tuple val(meta5), path(addprofile)
    tuple val(meta6), path(addlong)
```

File: modules/nf-core/mafft/tests/main.nf.test (hunk 4 above)
Comment (Joon-Klaps) [reply modules/nf-core/mafft/tests/main.nf.test:13]: Was for a faster update snapshot and forgot to uncomment as I had been trying multiple ways to do it

**Review by sateeshperi — APPROVED** [review]

**Pull request merged**